### PR TITLE
Add popoverControllerShouldDismissPopover

### PR DESCRIPTION
**Pull request by Vanilil**
Add UIPopoverControllerDelegate method popoverControllerShouldDismissPopover to dismiss picker if we tap outside the picker and not on done or cancel button

### CODE REVIEW

**Comment by aljlue** [comment]
@skywinder Please approve of this fix. Without it, this is causing a memory leak and reproducable crashes!!!

File: Pickers/AbstractActionSheetPicker.m [hunk 1]
```diff
@@ -779,6 +779,12 @@ - (void)presentPopover:(UIPopoverController *)popover {
 
 #pragma mark - Popoverdelegate
 
+- (BOOL)popoverControllerShouldDismissPopover:(UIPopoverController *)popoverController
+{
+    [self dismissPicker];
+    return YES;
```
Comment (skywinder) [inline Pickers/AbstractActionSheetPicker.m]: which case should it solve? I'm not sure, that we have to put in in abstreact class.

**Comment by skywinder** [comment]
in addition to this fix I think we have to wrap deprecated function : `popoverControllerDidDismissPopover` as well

**Comment by skywinder** [comment]
Hello, please fix this small thing (or update it from dvelop branch and I will merge it soon!

**Pull request closed without merging**